### PR TITLE
adding 'execute' to BloqadeSchema

### PR DESCRIPTION
**Pull request by weinbe58**
simple PR, there is an example in the test/execute.jl the function gives reasonable output but given that the output is random based on sampling the wave function, a test for the exact output might be tricky.

### CODE REVIEW

File: lib/BloqadeSchema/src/execute.jl
Comment (weinbe58) [inline lib/BloqadeSchema/src/execute.jl:13]: One question I have is whether or not this function will give a clear error in the event of a schema json that isn't formatted correctly.

File: lib/BloqadeSchema/src/execute.jl
Comment (Roger-luo) [inline lib/BloqadeSchema/src/execute.jl:12]: I think we probably should have three overloads of `execute` for convenience:

```julia
execute(json::Dict)
execute(json::String)
execute(schema::TaskSepcification)
```

since sometimes the schema is already parsed (e.g someone tweaking an existing schema via a `Dict` or `TaskSpecification`). And I think the dispatch here would also be strictly static, so it should be fine.

File: lib/BloqadeSchema/src/execute.jl
Comment (Roger-luo) [reply lib/BloqadeSchema/src/execute.jl:13]: what do you mean by formatted correctly?

**Review by Roger-luo — CHANGES_REQUESTED** [review]
only need a minor change, LGTM in general

File: lib/BloqadeSchema/src/execute.jl
Comment (weinbe58) [reply lib/BloqadeSchema/src/execute.jl:12]: sounds good, I was thinking the same thing.

File: lib/BloqadeSchema/src/execute.jl
Comment (Roger-luo) [reply lib/BloqadeSchema/src/execute.jl:13]: I see, I believe that will cause `from_dict` throwing an error? since fields may in place based on the schema specification defined by `TaskSpecification`

File: lib/BloqadeSchema/src/execute.jl
Comment (Roger-luo) [inline lib/BloqadeSchema/src/execute.jl:21]:
```suggestion
function execute(dict::AbstractDict{String})
```

File: lib/BloqadeSchema/src/execute.jl
Comment (Roger-luo) [inline lib/BloqadeSchema/src/execute.jl:27]:
```suggestion
```

**Review by Roger-luo — CHANGES_REQUESTED** [review]
you don't need to depend on `OrderedDict` here, just use `AbstractDict` would be fine. We also want to limit the key to be `String` as that's the only allowed value in `Configurations` mapped types.

File: lib/BloqadeSchema/Project.toml
Comment (Roger-luo) [inline lib/BloqadeSchema/Project.toml:14]:
```suggestion
```

File: lib/BloqadeSchema/src/BloqadeSchema.jl
Comment (Roger-luo) [inline lib/BloqadeSchema/src/BloqadeSchema.jl:12]:
```suggestion
```

**Review by Roger-luo — APPROVED** [review]
just need to wait for CI to pass

**Pull request merged**
